CitationStyleLanguagePlugin.inc.php

```diff
@@ -299,23 +317,17 @@ public function getArticleTemplateData($hookName, $args) {
 	 * @see Zotero's mappings https://aurimasv.github.io/z2csl/typeMap.xml#map-journalArticle
 	 * @see Mendeley's mappings http://support.mendeley.com/customer/portal/articles/364144-csl-type-mapping
 	 * @param $request Request
-	 * @param $article Submission
+	 * @param $submission Submission
 	 * @param $citationStyle string Name of the citation style to use.
-	 * @param $issue Issue Optional. Will fetch from db if not passed.
-	 * @param $publication Publication Optional. A particular version
 	 * @return string
 	 */
-	public function getCitation($request, $article, $citationStyle = 'apa', $issue = null, $publication = null) {
-		$publication = $publication ?? $article->getCurrentPublication();
-		$issueDao = DAORegistry::getDAO('IssueDAO'); /* @var $issueDao IssueDAO */
-		$issue = $issue ?? $issueDao->getById($publication->getData('issueId'));
+	public function getCitation($request, $submission, $citationStyle = 'apa', $issue, $submissionNoun) {
```

Comment:
Instead of passing in the `$submissionNoun` as an argument, can you split the check into a protected method and use that wherever you need it. Something like:

```php
protected function getPublicationTypeUrlPath(): string
{
    switch ($this->applicationName) {
        case 'ojs2': return 'article';
        case 'ops': return 'preprint';
    }
    return '';
}
```

Reply:
Also, keep `$issue = null` in the function signature, so that it isn't required.